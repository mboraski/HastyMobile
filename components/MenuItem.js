// Third Party Imports
import React, { Component } from 'react';
import { 
    StyleSheet,
    View,
    Image,
    Text
} from 'react-native';

// Relative Imports
import { emY } from '../utils/em';
import Color from '../constants/Color';

const SIZE = emY(1.44);
const BADGE_SIZE = emY(1.69);

type Props = {
    image: any,
    title: string,
    badge: string
};

class MenuItem extends Component {
    props: Props;

    render() {
        const { image, title, badge } = this.props;
        let badgeElement = null;

        if (badge) {
            badgeElement = (
                <View style={styles.badgeContainer}>
                    <Text style={styles.badge}>{badge}</Text>   
                </View>
            );
        }
        return (
            <View style={styles.container}>
                <Image style={styles.image} source={image} />  
                <Text style={styles.title}>{title}</Text> 
                { badgeElement }
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        alignItems: 'center',
        flexDirection: 'row',
        height: emY(3.5),
        borderBottomWidth: 1,
        borderBottomColor: Color.GREY_200
    },
    image: {
        maxWidth: SIZE,
        maxHeight: SIZE,
        resizeMode: 'contain',
        marginRight: emY(1.4)
    },
    title: {
        fontSize: emY(0.831),
        color: Color.GREY_700,
        textAlign: 'center'
    },
    badgeContainer: {
        position: 'absolute',
        alignItems: 'center',
        justifyContent: 'center',
        right: emY(1.2),
        width: BADGE_SIZE,
        height: BADGE_SIZE,
        borderRadius: BADGE_SIZE / 2,
        backgroundColor: Color.GREY_300,
    },
    badge: {
        backgroundColor: 'transparent',
        color: Color.WHITE,
        fontSize: emY(0.83)
    }
});

export default MenuItem;
