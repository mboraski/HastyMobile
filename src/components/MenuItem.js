// Third Party Imports
import React, { Component } from 'react';
import { StyleSheet, Image, Text, TouchableOpacity } from 'react-native';

// Relative Imports
import { emY } from '../utils/em';
import Color from '../constants/Color';

const SIZE = emY(1.44);
const BADGE_SIZE = emY(1.69);

type Props = {
    image: any,
    title: string,
    badge: string
};

class MenuItem extends Component {
    onPress = () => {
        const { route, activeItemKey, onPress } = this.props;
        const focused = route && activeItemKey === route.key;
        onPress({ route, focused });
    };

    props: Props;

    render() {
        const { image, title, badge, ...props } = this.props;
        let badgeElement = null;

        if (badge) {
            badgeElement = (
                <TouchableOpacity {...props} onPress={this.onPress} style={styles.badgeContainer}>
                    <Text style={styles.badge}>{badge}</Text>
                </TouchableOpacity>
            );
        }
        return (
            <TouchableOpacity {...props} onPress={this.onPress} style={styles.container}>
                <Image style={styles.image} source={image} />
                <Text style={styles.title}>{title}</Text>
                {badgeElement}
            </TouchableOpacity>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        alignItems: 'center',
        flexDirection: 'row',
        height: emY(3.5),
        borderBottomWidth: 1,
        borderBottomColor: Color.GREY_200
    },
    image: {
        maxWidth: SIZE,
        maxHeight: SIZE,
        resizeMode: 'contain',
        marginRight: emY(1.4)
    },
    title: {
        fontSize: emY(0.831),
        color: Color.GREY_700,
        textAlign: 'center'
    },
    badgeContainer: {
        position: 'absolute',
        alignItems: 'center',
        justifyContent: 'center',
        right: emY(1.2),
        width: BADGE_SIZE,
        height: BADGE_SIZE,
        borderRadius: BADGE_SIZE / 2,
        backgroundColor: Color.GREY_300
    },
    badge: {
        backgroundColor: 'transparent',
        color: Color.WHITE,
        fontSize: emY(0.83)
    }
});

export default MenuItem;
