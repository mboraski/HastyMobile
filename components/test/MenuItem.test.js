import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import MenuItem from '../MenuItem';
import notificationIcon from '../../assets/icons/notification.png';

const initialState = {};

describe('MenuItem', () => {
    const middlewares = [];
    const mockStore = configureStore(middlewares);
    it('renders correctly', () => {
        const wrapper = shallow(<MenuItem 
            image={notificationIcon} title="Notifications" badge="3" 
        />, {
            context: { store: mockStore(initialState) }
        });
        const render = wrapper.dive();
        expect(render).toMatchSnapshot();
    });
});
