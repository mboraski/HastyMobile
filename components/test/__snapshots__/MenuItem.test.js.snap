// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <AnimatedComponent
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
        Array [
            Object {
              "alignItems": "center",
              "borderBottomColor": "#E4E4E4",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "height": 56,
            },
            Object {
              "opacity": 1,
            },
          ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
>
    <Image
        source={1}
        style={
            Object {
                "marginRight": 22.4,
                "maxHeight": 23.04,
                "maxWidth": 23.04,
                "resizeMode": "contain",
              }
        }
    />
    <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
            Object {
                "color": "#4c4c4c",
                "fontSize": 13.296,
                "textAlign": "center",
              }
        }
    >
        Notifications
    </Text>
    <TouchableOpacity
        activeOpacity={0.2}
        focusedOpacity={0.7}
        onPress={[Function]}
        style={
            Object {
                "alignItems": "center",
                "backgroundColor": "#cccccc",
                "borderRadius": 13.52,
                "height": 27.04,
                "justifyContent": "center",
                "position": "absolute",
                "right": 19.2,
                "width": 27.04,
              }
        }
    >
        <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
                Object {
                    "backgroundColor": "transparent",
                    "color": "#ffffff",
                    "fontSize": 13.28,
                  }
            }
        >
            3
        </Text>
    </TouchableOpacity>
</AnimatedComponent>,
  "nodes": Array [
    <AnimatedComponent
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
            Array [
                  Object {
                    "alignItems": "center",
                    "borderBottomColor": "#E4E4E4",
                    "borderBottomWidth": 1,
                    "flexDirection": "row",
                    "height": 56,
                  },
                  Object {
                    "opacity": 1,
                  },
                ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
>
      <Image
            source={1}
            style={
                  Object {
                        "marginRight": 22.4,
                        "maxHeight": 23.04,
                        "maxWidth": 23.04,
                        "resizeMode": "contain",
                      }
            }
      />
      <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
                  Object {
                        "color": "#4c4c4c",
                        "fontSize": 13.296,
                        "textAlign": "center",
                      }
            }
      >
            Notifications
      </Text>
      <TouchableOpacity
            activeOpacity={0.2}
            focusedOpacity={0.7}
            onPress={[Function]}
            style={
                  Object {
                        "alignItems": "center",
                        "backgroundColor": "#cccccc",
                        "borderRadius": 13.52,
                        "height": 27.04,
                        "justifyContent": "center",
                        "position": "absolute",
                        "right": 19.2,
                        "width": 27.04,
                      }
            }
      >
            <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                        Object {
                              "backgroundColor": "transparent",
                              "color": "#ffffff",
                              "fontSize": 13.28,
                            }
                  }
            >
                  3
            </Text>
      </TouchableOpacity>
</AnimatedComponent>,
  ],
  "options": Object {
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
      },
      "_currentElement": <TouchableOpacity
        activeOpacity={0.2}
        focusedOpacity={0.7}
        onPress={[Function]}
        style={
                Object {
                        "alignItems": "center",
                        "borderBottomColor": "#E4E4E4",
                        "borderBottomWidth": 1,
                        "flexDirection": "row",
                        "height": 56,
                      }
        }
>
        <Image
                source={1}
                style={
                        Object {
                                "marginRight": 22.4,
                                "maxHeight": 23.04,
                                "maxWidth": 23.04,
                                "resizeMode": "contain",
                              }
                }
        />
        <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                        Object {
                                "color": "#4c4c4c",
                                "fontSize": 13.296,
                                "textAlign": "center",
                              }
                }
        >
                Notifications
        </Text>
        <TouchableOpacity
                activeOpacity={0.2}
                focusedOpacity={0.7}
                onPress={[Function]}
                style={
                        Object {
                                "alignItems": "center",
                                "backgroundColor": "#cccccc",
                                "borderRadius": 13.52,
                                "height": 27.04,
                                "justifyContent": "center",
                                "position": "absolute",
                                "right": 19.2,
                                "width": 27.04,
                              }
                }
        >
                <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                                Object {
                                        "backgroundColor": "transparent",
                                        "color": "#ffffff",
                                        "fontSize": 13.28,
                                      }
                        }
                >
                        3
                </Text>
        </TouchableOpacity>
</TouchableOpacity>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance":  {
        "_cancelLongPressDelayTimeout": [Function],
        "_endHighlight": [Function],
        "_getChildStyleOpacityWithDefault": [Function],
        "_getDistanceBetweenPoints": [Function],
        "_handleDelay": [Function],
        "_handleLongDelay": [Function],
        "_handleQueryLayout": [Function],
        "_isHighlight": [Function],
        "_opacityActive": [Function],
        "_opacityFocused": [Function],
        "_opacityInactive": [Function],
        "_performSideEffectsForTransition": [Function],
        "_reactInternalInstance": [Circular],
        "_receiveSignal": [Function],
        "_remeasureMetricsOnActivation": [Function],
        "_savePressInLocation": [Function],
        "_startHighlight": [Function],
        "blur": [Function],
        "cancelAnimationFrame": [Function],
        "clearImmediate": [Function],
        "clearInterval": [Function],
        "clearTimeout": [Function],
        "context": Object {},
        "focus": [Function],
        "measure": [Function],
        "measureInWindow": [Function],
        "measureLayout": [Function],
        "props": Object {
          "activeOpacity": 0.2,
          "children": Array [
            <Image
              source={1}
              style={
                            Object {
                                          "marginRight": 22.4,
                                          "maxHeight": 23.04,
                                          "maxWidth": 23.04,
                                          "resizeMode": "contain",
                                        }
              }
/>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                            Object {
                                          "color": "#4c4c4c",
                                          "fontSize": 13.296,
                                          "textAlign": "center",
                                        }
              }
>
              Notifications
</Text>,
            <TouchableOpacity
              activeOpacity={0.2}
              focusedOpacity={0.7}
              onPress={[Function]}
              style={
                            Object {
                                          "alignItems": "center",
                                          "backgroundColor": "#cccccc",
                                          "borderRadius": 13.52,
                                          "height": 27.04,
                                          "justifyContent": "center",
                                          "position": "absolute",
                                          "right": 19.2,
                                          "width": 27.04,
                                        }
              }
>
              <Text
                            accessible={true}
                            allowFontScaling={true}
                            ellipsizeMode="tail"
                            style={
                                          Object {
                                                        "backgroundColor": "transparent",
                                                        "color": "#ffffff",
                                                        "fontSize": 13.28,
                                                      }
                            }
              >
                            3
              </Text>
</TouchableOpacity>,
          ],
          "focusedOpacity": 0.7,
          "onPress": [Function],
          "style": Object {
            "alignItems": "center",
            "borderBottomColor": "#E4E4E4",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "height": 56,
          },
        },
        "refs": Object {},
        "requestAnimationFrame": [Function],
        "setImmediate": [Function],
        "setInterval": [Function],
        "setNativeProps": [Function],
        "setOpacityTo": [Function],
        "setTimeout": [Function],
        "state": Object {
          "anim": 1,
          "touchable": Object {
            "responderID": null,
            "touchState": undefined,
          },
        },
        "touchableGetHighlightDelayMS": [Function],
        "touchableGetHitSlop": [Function],
        "touchableGetInitialState": [Function],
        "touchableGetLongPressDelayMS": [Function],
        "touchableGetPressOutDelayMS": [Function],
        "touchableGetPressRectOffset": [Function],
        "touchableHandleActivePressIn": [Function],
        "touchableHandleActivePressOut": [Function],
        "touchableHandleLongPress": [Function],
        "touchableHandlePress": [Function],
        "touchableHandleResponderGrant": [Function],
        "touchableHandleResponderMove": [Function],
        "touchableHandleResponderRelease": [Function],
        "touchableHandleResponderTerminate": [Function],
        "touchableHandleResponderTerminationRequest": [Function],
        "touchableHandleStartShouldSetResponder": [Function],
        "touchableLongPressCancelsPress": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <AnimatedComponent
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
                    Array [
                              Object {
                                "alignItems": "center",
                                "borderBottomColor": "#E4E4E4",
                                "borderBottomWidth": 1,
                                "flexDirection": "row",
                                "height": 56,
                              },
                              Object {
                                "opacity": 1,
                              },
                            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
>
          <Image
                    source={1}
                    style={
                              Object {
                                        "marginRight": 22.4,
                                        "maxHeight": 23.04,
                                        "maxWidth": 23.04,
                                        "resizeMode": "contain",
                                      }
                    }
          />
          <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                              Object {
                                        "color": "#4c4c4c",
                                        "fontSize": 13.296,
                                        "textAlign": "center",
                                      }
                    }
          >
                    Notifications
          </Text>
          <TouchableOpacity
                    activeOpacity={0.2}
                    focusedOpacity={0.7}
                    onPress={[Function]}
                    style={
                              Object {
                                        "alignItems": "center",
                                        "backgroundColor": "#cccccc",
                                        "borderRadius": 13.52,
                                        "height": 27.04,
                                        "justifyContent": "center",
                                        "position": "absolute",
                                        "right": 19.2,
                                        "width": 27.04,
                                      }
                    }
          >
                    <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                        Object {
                                                  "backgroundColor": "transparent",
                                                  "color": "#ffffff",
                                                  "fontSize": 13.28,
                                                }
                              }
                    >
                              3
                    </Text>
          </TouchableOpacity>
</AnimatedComponent>,
        "_debugID": 4,
        "_renderedOutput": <AnimatedComponent
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
                    Array [
                              Object {
                                "alignItems": "center",
                                "borderBottomColor": "#E4E4E4",
                                "borderBottomWidth": 1,
                                "flexDirection": "row",
                                "height": 56,
                              },
                              Object {
                                "opacity": 1,
                              },
                            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
>
          <Image
                    source={1}
                    style={
                              Object {
                                        "marginRight": 22.4,
                                        "maxHeight": 23.04,
                                        "maxWidth": 23.04,
                                        "resizeMode": "contain",
                                      }
                    }
          />
          <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                              Object {
                                        "color": "#4c4c4c",
                                        "fontSize": 13.296,
                                        "textAlign": "center",
                                      }
                    }
          >
                    Notifications
          </Text>
          <TouchableOpacity
                    activeOpacity={0.2}
                    focusedOpacity={0.7}
                    onPress={[Function]}
                    style={
                              Object {
                                        "alignItems": "center",
                                        "backgroundColor": "#cccccc",
                                        "borderRadius": 13.52,
                                        "height": 27.04,
                                        "justifyContent": "center",
                                        "position": "absolute",
                                        "right": 19.2,
                                        "width": 27.04,
                                      }
                    }
          >
                    <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                        Object {
                                                  "backgroundColor": "transparent",
                                                  "color": "#ffffff",
                                                  "fontSize": 13.28,
                                                }
                              }
                    >
                              3
                    </Text>
          </TouchableOpacity>
</AnimatedComponent>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <TouchableOpacity
    activeOpacity={0.2}
    focusedOpacity={0.7}
    onPress={[Function]}
    style={
        Object {
            "alignItems": "center",
            "borderBottomColor": "#E4E4E4",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "height": 56,
          }
    }
>
    <Image
        source={1}
        style={
            Object {
                "marginRight": 22.4,
                "maxHeight": 23.04,
                "maxWidth": 23.04,
                "resizeMode": "contain",
              }
        }
    />
    <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
            Object {
                "color": "#4c4c4c",
                "fontSize": 13.296,
                "textAlign": "center",
              }
        }
    >
        Notifications
    </Text>
    <TouchableOpacity
        activeOpacity={0.2}
        focusedOpacity={0.7}
        onPress={[Function]}
        style={
            Object {
                "alignItems": "center",
                "backgroundColor": "#cccccc",
                "borderRadius": 13.52,
                "height": 27.04,
                "justifyContent": "center",
                "position": "absolute",
                "right": 19.2,
                "width": 27.04,
              }
        }
    >
        <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
                Object {
                    "backgroundColor": "transparent",
                    "color": "#ffffff",
                    "fontSize": 13.28,
                  }
            }
        >
            3
        </Text>
    </TouchableOpacity>
</TouchableOpacity>,
}
`;
